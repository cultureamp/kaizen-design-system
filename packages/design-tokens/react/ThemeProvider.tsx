import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
