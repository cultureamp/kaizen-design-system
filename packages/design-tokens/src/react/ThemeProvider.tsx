import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
