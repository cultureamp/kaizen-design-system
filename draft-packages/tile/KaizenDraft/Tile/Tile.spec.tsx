import * as React from "react"

import {
  InformationTile,
  MultiActionTile,
  TileAction,
  TileInformation,
} from "@kaizen/draft-tile"
import { cleanup, render } from "@testing-library/react"

const primaryAction: TileAction = {
  label: "View",
  href: "href",
}

const secondaryAction: TileAction = {
  label: "Preview",
  href: "href2",
}

const information: TileInformation = {
  text:
    "Additional information can be included on the underside of the tile and viewed on click of the information" +
    " icon.",
  secondaryAction: {
    label: "Learn more",
    href: "",
  },
}

afterEach(cleanup)

describe("<InformationTile />", () => {
  describe("snapshots", () => {
    it("renders InformationTile", () => {
      const { container } = render(
        <InformationTile
          title="Title"
          metadata="Metadata"
          footer={<div>Hello world</div>}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it("renders InformationTile with children", () => {
      const { container } = render(
        <InformationTile
          title="Title"
          metadata="Metadata"
          footer={<div>Hello world</div>}
        >
          <div>Child content</div>
        </InformationTile>
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it("renders InformationTile with information data", () => {
      const { container } = render(
        <InformationTile
          title="Title"
          metadata="Metadata"
          footer={<div>Hello world</div>}
          information={information}
        >
          <div>Child content</div>
        </InformationTile>
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})

describe("<MultiActionTile />", () => {
  describe("snapshots", () => {
    it("renders MultiActionTile", () => {
      const { container } = render(
        <MultiActionTile
          title="Title"
          metadata="Metadata"
          primaryAction={primaryAction}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it("renders MultiActionTile with secondary action", () => {
      const { container } = render(
        <MultiActionTile
          title="Title"
          metadata="Metadata"
          primaryAction={primaryAction}
          secondaryAction={secondaryAction}
        />
      )

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
