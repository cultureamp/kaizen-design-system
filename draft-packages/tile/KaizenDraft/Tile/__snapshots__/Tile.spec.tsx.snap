// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationTile /> snapshots renders InformationTile 1`] = `
<div
  class="root"
>
  <div
    class="tile"
  >
    <div
      class="face faceFront"
    >
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="children"
      />
      <div
        class="footer"
      >
        <div>
          Hello world
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<InformationTile /> snapshots renders InformationTile with children 1`] = `
<div
  class="root"
>
  <div
    class="tile"
  >
    <div
      class="face faceFront"
    >
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="children"
      >
        <div>
          Child content
        </div>
      </div>
      <div
        class="footer"
      >
        <div>
          Hello world
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<InformationTile /> snapshots renders InformationTile with information data 1`] = `
<div
  class="root"
>
  <div
    class="tile"
  >
    <div
      class="face faceFront"
    >
      <div
        class="informationBtn"
      >
        <span
          class="container"
        >
          <button
            aria-label="Information"
            class="button iconButton"
            title="Information"
            type="button"
          >
            <span
              class="content"
            >
              <span
                class="iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="80284680-3304-5869-9788-7428e6c6ee86"
                  class="icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 0 0"
                >
                  <use
                    xlink:href="#my-svg"
                  />
                </svg>
              </span>
            </span>
          </button>
        </span>
      </div>
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="children"
      >
        <div>
          Child content
        </div>
      </div>
      <div
        class="footer"
      >
        <div>
          Hello world
        </div>
      </div>
    </div>
    <div
      class="face faceBack"
    >
      <div
        class="informationBtn"
      >
        <span
          class="container"
        >
          <button
            aria-label="Information"
            class="button iconButton"
            disabled=""
            title="Information"
            type="button"
          >
            <span
              class="content"
            >
              <span
                class="iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  aria-labelledby="80284680-3304-5869-9788-7428e6c6ee86"
                  class="icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 0 0"
                >
                  <use
                    xlink:href="#my-svg"
                  />
                </svg>
              </span>
            </span>
          </button>
        </span>
      </div>
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="information"
      >
        <p
          class="paragraph body dark"
        >
          Additional information can be included on the underside of the tile and viewed on click of the information icon.
        </p>
        <div
          class="footer"
        >
          <div
            class="actions"
          >
            <span
              class="container"
            >
              <button
                class="button secondary"
                disabled=""
                type="button"
              >
                <span
                  class="content"
                >
                  <span
                    class="label"
                  >
                    Learn more
                  </span>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MultiActionTile /> snapshots renders MultiActionTile 1`] = `
<div
  class="root"
>
  <div
    class="tile"
  >
    <div
      class="face faceFront"
    >
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="children"
      />
      <div
        class="footer"
      >
        <div
          class="actions"
        >
          <span
            class="container"
          >
            <a
              class="button"
              href="href"
              target="_self"
            >
              <span
                class="content"
              >
                <span
                  class="label"
                >
                  View
                </span>
              </span>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MultiActionTile /> snapshots renders MultiActionTile with secondary action 1`] = `
<div
  class="root"
>
  <div
    class="tile"
  >
    <div
      class="face faceFront"
    >
      <div
        class="title"
      >
        <h3
          class="heading heading-3 dark small"
        >
          Title
        </h3>
        <div
          class="pt-0-point-25 m-0"
        >
          <p
            class="paragraph small dark-reduced-opacity"
          >
            Metadata
          </p>
        </div>
      </div>
      <div
        class="children"
      />
      <div
        class="footer"
      >
        <div
          class="actions"
        >
          <span
            class="container"
          >
            <a
              class="button secondary"
              href="href2"
              target="_self"
            >
              <span
                class="content"
              >
                <span
                  class="label"
                >
                  Preview
                </span>
              </span>
            </a>
          </span>
          <span
            class="container"
          >
            <a
              class="button"
              href="href"
              target="_self"
            >
              <span
                class="content"
              >
                <span
                  class="label"
                >
                  View
                </span>
              </span>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
